src/app/services/client/course.ts

```diff
@@ -0,0 +1,89 @@
+import { db } from "../../../../firebase";
+import { Course } from "@interfaces/db";
+import { courseConverter } from "../firestore";
+import {
+  deleteDoc,
+  doc,
+  DocumentReference,
+  PartialWithFieldValue,
+  setDoc,
+  updateDoc,
+} from "firebase/firestore";
+
+type addCourse = Pick<Course, "name" | "location" | "professors"> &
+PartialWithFieldValue<Course>
+
+export const addCourse = async (
+  course: addCourse,
+  courseID: string,
+) => {
+  try {
+    const courseDoc: DocumentReference = doc(
+      db,
+      `courses/${courseID}`
+    ).withConverter(courseConverter);
+
+    const newCourse: addCourse = {
+      name: course.name,
+      location: course.location,
+      professors: course.professors,
+      students: course.students ?? [],
+      tas: course.tas ?? [],
+      onDuty: course.onDuty ?? [],
+      tags: course.tags ?? {},
+    };
+
+    await setDoc(courseDoc, newCourse);
+
+    console.log("New course created: ");
+    return courseDoc;
+  } catch (error) {
```

Comment:
If we get the reference on the addDoc `const newCourseDoc = await addDoc(courseDoc, newCourse);`

We can return `{...newCourse, courseId: newCourseDoc.id}`

Reply:
If I understand you correctly, do you mean keep ID in the interface, but do not add it to documents on Firebase?